skabase/SKALogger/test/SKALogger_test.py

```diff
@@ -52,170 +47,153 @@ def mocking(cls):
+        levels.append(logging_level)
+        targets = []
+        targets.append(logging_target)
+        device_details = []
+        device_details.append(levels)
+        device_details.append(targets)
+        tango_context.device.SetElementLoggingLevel(device_details)
+        dev_proxy = DeviceProxy(logging_target)
+        assert dev_proxy.elementLoggingLevel == logging_level
         # PROTECTED REGION END #    //  SKALogger.test_SetElementLoggingLevel
 
-    # PROTECTED REGION ID(SKALogger.test_SetStorageLoggingLevel_decorators) ENABLED START #
-    # PROTECTED REGION END #    //  SKALogger.test_SetStorageLoggingLevel_decorators
-    def test_SetStorageLoggingLevel(self, tango_context):
+    @pytest.mark.parametrize("logging_level", [3])
+    @pytest.mark.parametrize("logging_target", ["logger/test/1"])
+    def test_SetStorageLoggingLevel(self, tango_context, logging_level, logging_target):
         """Test for SetStorageLoggingLevel"""
         # PROTECTED REGION ID(SKALogger.test_SetStorageLoggingLevel) ENABLED START #
-        assert tango_context.device.SetStorageLoggingLevel("") == None
+        levels = []
+        levels.append(logging_level)
+        targets = []
+        targets.append(logging_target)
+        device_details = []
+        device_details.append(levels)
+        device_details.append(targets)
+        tango_context.device.SetStorageLoggingLevel(device_details)
+        dev_proxy = DeviceProxy(logging_target)
+        assert dev_proxy.storageLoggingLevel == logging_level
         # PROTECTED REGION END #    //  SKALogger.test_SetStorageLoggingLevel
 
-    # PROTECTED REGION ID(SKALogger.test_GetMetrics_decorators) ENABLED START #
-    # PROTECTED REGION END #    //  SKALogger.test_GetMetrics_decorators
-    def test_GetMetrics(self, tango_context):
-        """Test for GetMetrics"""
-        # PROTECTED REGION ID(SKALogger.test_GetMetrics) ENABLED START #
-        assert tango_context.device.GetMetrics() == ""
-        # PROTECTED REGION END #    //  SKALogger.test_GetMetrics
-
-    # PROTECTED REGION ID(SKALogger.test_ToJson_decorators) ENABLED START #
-    # PROTECTED REGION END #    //  SKALogger.test_ToJson_decorators
-    def test_ToJson(self, tango_context):
-        """Test for ToJson"""
-        # PROTECTED REGION ID(SKALogger.test_ToJson) ENABLED START #
-        assert tango_context.device.ToJson("") == ""
-        # PROTECTED REGION END #    //  SKALogger.test_ToJson
-
-    # PROTECTED REGION ID(SKALogger.test_GetVersionInfo_decorators) ENABLED START #
-    # PROTECTED REGION END #    //  SKALogger.test_GetVersionInfo_decorators
     def test_GetVersionInfo(self, tango_context):
         """Test for GetVersionInfo"""
         # PROTECTED REGION ID(SKALogger.test_GetVersionInfo) ENABLED START #
-        assert tango_context.device.GetVersionInfo() == [""]
+        versionPattern = re.compile(
+            r'SKALogger, tangods-skabasedevice, [0-9].[0-9].[0-9], A generic base device for SKA')
```

Comment:
A Logger device shouldn't report that it is a "tangods-skabasedevice", but that is what the current implementation is doing.

Would have to add something like the base class has here https://github.com/ska-telescope/lmc-base-classes/blob/c6484ea/skabase/SKABaseDevice/SKABaseDevice/SKABaseDevice.py#L287-L289
to the Logger's `init_device` method.

This is a general problem affecting all the derived classes, so best to add it to the backlog instead of changing in this PR.  Same for `versionId` and `buildState`

Reply:
Yeah, will keep this in backlog.

Reply:
Have corrected here for Logger class.